### ISSUE TITLE
CLI argument to specify the number of `dns worker threads` in `fetch`

### ISSUE THREAD

**Issue by niklasad1** [paritytech/parity-ethereum#9623]
Add functionality to control the number of `DNS worker threads` in [fetch](https://github.com/paritytech/parity-ethereum/tree/master/util/fetch), it is currently hardcoded to 4

This is the line that specifies the number of `DNS worker threads`:
https://github.com/paritytech/parity-ethereum/blob/master/util/fetch/src/client.rs#L211

**Comment by debris** [comment]
is it really needed? a difference between footprint of a single thread and a few threads in this case is probably so small that no user will bother to change this CLI param

**Comment by debris** [comment]
@tomusdrw what do you think?

**Comment by niklasad1** [comment]
@debris this is part of an attempt to make the light client `lighter` and AFAIK the default-stack size for pthread is somewhere between 2MB - 32 MB (on x86) not sure on other platforms and Windows though.

In the light-client mode I would say that this is worth the effort!

**Comment by debris** [comment]
so maybe let's just set it by default to 1 thread for `light` client and keep 4 for normal run :)

**Comment by tomusdrw** [comment]
I second @debris, 1 thread for light client and 4 for full node seems like a good idea. Possilby we could even have always 1 thread, since we don't really talk to random addreses at all - anything that get's resolved on a first request is probably cached forever.

**Comment by niklasad1** [comment]
Sure, it sounds good to me!

However, because we compile light and full the same I think a simple flag to ```fetch``` is the way to go here for now and I'm not a big fan of feature flags!

**Comment by tomusdrw** [comment]
@niklasad1 we can check the `--light` flag, no? And run with 1 thread if that's enabled.

**Comment by niklasad1** [comment]
@tomusdrw 
Yeah, I meant in `parity::run::excute` we check the light flag either to run as full or light. In those different methods, we need to pass this flag into the ```fetch::Client``` constructor whether if it `should` be light or not! So, yes we could use the light flag or simple bool!

I submit a PR to this probably easier to understand then!
